mapie/conformity_scores/residual_conformity_scores.py

```diff
@@ -132,3 +141,275 @@ def get_estimation_distribution(
         """
         self._check_predicted_data(y_pred)
         return np.multiply(y_pred, np.add(1, conformity_scores))
+
+
+class FittedResidualNormalisingScore(ConformityScore):
+    # or ConformalResiudalFittingScore
+    # or NormalizedScore
+    """
+    FittedResidualNormalisingScore (also called CRF) score.
+
+    The signed conformity score = (y - y_pred) / r_pred. r_pred being the
+    predicted residual (|y - y_pred|) of the base estimator.
+    It is calculated by a model that learns to predict these residuals.
+    The learning is done with the log of the residual and we use the
+    exponential of the prediction to avoid negative values.
+
+    The conformity score is symmetrical and allows the calculation of adaptive
+    prediction intervals (taking X into account). It is possible to use it
+    only with split and prefit methods (not with cross methods).
+
+    Warning : if the estimator provided is not fitted a subset of the
+    calibration data will be used to fit the model (50% by default).
+
+    Parameters
+    ----------
+    residual_estimator: Optional[RegressorMixin]
+        The model that learns to predict the residuals of the base estimator.
+        It can be any regressor with scikit-learn API (i.e. with ``fit``
+        and ``predict`` methods).
+        If ``None``, estimator defaults to a ``LinearRegression`` instance.
+
+    prefit: bool
+        Specify if the residual_estimator is already fitted or not.
+        By default ``False``.
+
+    split_size: Optional[Union[int, float]]
+        The proportion of data that is used to fit the residual_estimator
+        By default 0.5.
+
+    random_state: 0ptional[Union[int, np.random.RandomState]]
+        Pseudo random number used for random sampling.
+        Pass an int for reproducible output across multiple function calls.
+        By default ``None``.
+    """
+
+    def __init__(
+        self,
+        residual_estimator: Optional[RegressorMixin] = None,
+        prefit: bool = False,
+        split_size: Optional[Union[int, float]] = None,
+        random_state: Optional[Union[int, np.random.RandomState]] = None
+    ) -> None:
+        super().__init__(sym=True, consistency_check=False)
+        self.prefit = prefit
+        self.residual_estimator = residual_estimator
+        self.split_size = split_size
+        self.random_state = random_state
+
```

Comment:
```suggestion
    def __init__(
        self,
        residual_estimator: Optional[RegressorMixin] = None,
        prefit: bool = False,
        split_size: Optional[Union[int, float]] = None,
        random_state: Optional[Union[int, np.random.RandomState]] = None,
        sym: bool = True,
        consistency_check: bool = False
    ) -> None:
        super().__init__(sym=sym, consistency_check=consistency_check)
        self.prefit = prefit
        self.residual_estimator = residual_estimator
        self.split_size = split_size
        self.random_state = random_state
```

Reply:
This is not done in other conformity score classes.

Reply:
I agree with you. But in a previous conversation, we concluded that it would be beneficial to give the choice (especially for `sym`). In any case, it doesn't change the default behaviour. It will be a good change if you add it.

Reply:
Okay, so I propose to add it for all of the score classes ?

Reply:
Your contribution is only for CRF. So just add your changes for CRF at this time.

Reply:
Done